src/toolkit.cpp

```diff
@@ -603,7 +609,18 @@ bool Toolkit::LoadData(const std::string &data)
         pugi::xml_document xmlfile;
         xmlfile.load_string(data.c_str());
         stringstream conversion;
+
+        // Temporarily redirect cerr:
+        std::stringstream captured_cerr;
+        std::streambuf *cerr_buf = std::cerr.rdbuf();
+        std::cerr.rdbuf(captured_cerr.rdbuf());
+
         bool status = converter.convert(conversion, xmlfile);
+        LogWarning(captured_cerr.str().c_str());
+
+        // Restore cerr:
+        std::cerr.rdbuf(cerr_buf);
+
```

Comment:
Since there is a lot of code repetition, just add and use some members and  methods (e.g., `Toolkit::LogRedirectStart` and `LogRedirectEnd`) to do this.

Reply:
👍🏻 Those functions currently do not exist, so I am presuming I would add them to the Toolkit class (as you explicitly say, so I am not really asking, just confirming :-)?

And the revised code in such cases will be:

```
LogRedirectStart();
bool status = converter.convert(conversion, xmlfile);
LogRedirectEnd();
```

Which looks much nicer.

And I presume that these would be hardwired to capture of `cerr` (and don't bother with `cout` as with the current code)?

Reply:
Implemented in commit https://github.com/rism-digital/verovio/pull/3767/commits/f79e85010724d42791bf4197257a056b79dbd7ce